### PR TITLE
Connect experiments page

### PR DESCRIPTION
**Pull request by smacker**
Implements functionality for #149.

Based on:
https://github.com/src-d/code-annotation/pull/152
https://github.com/src-d/code-annotation/pull/154

`mark as finished` link is commented, we don't have API for it now.

there are +2 queries to DB for each experiment, we can optimize it later if we need (if we will have like thousands of experiments). For now, I decided to keep it more flexible not performant.

### CODE REVIEW

File: server/repository/assignments.go [hunk 1]
```diff
@@ -24,6 +24,8 @@ const (
 	selectAssignmentsWhereExpPairSQL = `SELECT * FROM assignments WHERE experiment_id=$1 AND pair_id=$2`
 	updateAssignmentsSQL             = `UPDATE assignments SET answer='%v', duration=%v WHERE id=%v`
 	countPendingIDsSQL               = `SELECT count(id) FROM file_pairs WHERE experiment_id=$1 AND id NOT IN (SELECT pair_id FROM assignments WHERE experiment_id=$1 AND user_id=$2)`
+	countUserAssigmentsSQL           = `SELECT COUNT(*) FROM assignments WHERE experiment_id=$1 AND user_id=$1`
+	countCompleteUserAssigmentsSQL   = `SELECT COUNT(*) FROM assignments WHERE experiment_id=$1 AND user_id=$1 AND answer IS NOT null`
```
Comment (carlosms) [inline server/repository/assignments.go]: These two queries are using $1 for both arguments, user_id should be $2

File: server/repository/assignments.go [hunk 2]
```diff
@@ -152,3 +154,27 @@ func (repo *Assignments) Update(assignmentID int, answer string, duration int) e
 
 	return err
 }
+
+// CountUserAssigments returns number of assigments in given experiment for the given user
+func (repo *Assignments) CountUserAssigments(experimentID, userID int) (int, error) {
```
Comment (carlosms) [inline server/repository/assignments.go]: Typo in several places, Assigment -> assignment

**Review by carlosms — CHANGES_REQUESTED** [review]
I found a small bug, and typo

File: server/repository/assignments.go (hunk 1 above)
Comment (smacker) [reply server/repository/assignments.go]: thank you so much! fixed

File: server/repository/assignments.go (hunk 2 above)
Comment (smacker) [reply server/repository/assignments.go]: thank you! I never spell it correctly

**Comment by bzz** [comment]
@carlosms seems like all the feedback has been addressed now and it's ready for another pass.

**Review by bzz — APPROVED** [review]
LGTM

**Comment by bzz** [comment]
@carlosms do you think there still is something left before merging this?

**Comment by carlosms** [comment]
Right now there isn't a way to go to the dashboard. Upper left link takes you to `/exp/1/...` and the "dashboard" text in the upper right opens the dropdown, and in that drop down you can only click "review" and "logout".

I'm not sure if this problem should be solved in this PR, but it's here where I've noticed it.

**Comment by ricardobaeta** [comment]
@carlosms Thank you for the notice. I would suggest that the "logo/upper-left-link" takes people to the dashboard. And the "dashboard" link would change its label to "more" or "options" with the current dropdown "review/log out".

**Comment by smacker** [comment]
@carlosms it's solved in next PR. I kept dashboard unlinkable here on purpose because it has links to other experiments and those don't work yet. (I fixed it in next PR and dashboard is a default index page for logged in user)

**Review by carlosms — APPROVED** [review]
:+1:

**Pull request merged**